gobblin-api/src/main/java/org/apache/gobblin/configuration/State.java

```diff
@@ -62,16 +63,21 @@ public class State implements WritableShim {
   @Getter
   private Properties specProperties;
 
+  // This in-mem state will be used to share partition completion information across different MetadataWriter impls
+  public Map<String, Set<String>> tableToCompletedPartitions;
```

Comment:
Let's not put this map into a state object as this is one super common class in Gobblin and this map does not make sense for most of the use cases. Can you serialize the map and put it as one configuration of the properties?

Reply:
My intention is to make this map an in-memory transient object. I feel we shouldn't expose this as a config key that users can access and set externally. WDYT?

Reply:
The config can be set/overwritten on the flight at runtime, which is only for internally usage. Having a map does not make sense for most use cases of this state object, so I would suggest using the existing API.

Reply:
The thing is if I make it a config, we can basically set it manually outside of the code logic, which is not what we want, right? Given this is just some internal memory state.